core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FileBatchSource.java

```diff
@@ -17,66 +17,108 @@
-    "filename expansion (globbing) to read files.";
-
-  private final FileBatchConfig config;
+public class FileBatchSource extends AbstractFileSource {
+  public static final Schema DEFAULT_SCHEMA = PathTrackingInputFormat.getTextOutputSchema(null);
+  static final String INPUT_NAME_CONFIG = "input.path.name";
+  static final String INPUT_REGEX_CONFIG = "input.path.regex";
+  static final String LAST_TIME_READ = "last.time.read";
+  static final String CUTOFF_READ_TIME = "cutoff.read.time";
+  static final String USE_TIMEFILTER = "timefilter";
+  private static final Gson GSON = new Gson();
+  private static final Type ARRAYLIST_DATE_TYPE = new TypeToken<ArrayList<Date>>() { }.getType();
+  private final FileSourceConfig config;
 
-  public FileBatchSource(FileBatchConfig config) {
+  public FileBatchSource(FileSourceConfig config) {
     super(config);
     this.config = config;
   }
 
-  @VisibleForTesting
-  FileBatchConfig getConfig() {
-    return config;
-  }
+  @Override
+  public void prepareRun(BatchSourceContext context) throws Exception {
+    super.prepareRun(context);
 
-  /**
-   * Config class that contains all the properties needed for the file source.
-   */
-  public static class FileBatchConfig extends FileSourceConfig {
-    @Description(PATH_DESCRIPTION)
-    @Macro
-    public String path;
-
-    @VisibleForTesting
-    public FileBatchConfig() {
-      this(null, null, null, null, null, null, null, null, null, null, null, null, null);
+    // Need to create dataset now if macro was provided at configure time
+    if (config.getTimeTable() != null && !context.datasetExists(config.getTimeTable())) {
+      context.createDataset(config.getTimeTable(), KeyValueTable.class.getName(), DatasetProperties.EMPTY);
     }
+  }
 
-    public FileBatchConfig(String referenceName, @Nullable String fileRegex, @Nullable String timeTable,
-                           @Nullable String inputFormatClass, @Nullable String fileSystemProperties,
-                           @Nullable String format, @Nullable Long maxSplitSize,
-                           @Nullable Boolean ignoreNonExistingFolders, @Nullable Boolean recursive,
-                           String path, @Nullable String pathField,
-                           @Nullable Boolean fileNameOnly, @Nullable String schema) {
-      super(referenceName, fileRegex, timeTable, inputFormatClass, fileSystemProperties, format,
-            maxSplitSize, ignoreNonExistingFolders, recursive, pathField, fileNameOnly, schema);
-      this.path = path;
+  @Override
+  protected Map<String, String> getFileSystemProperties(BatchSourceContext context) {
+    Map<String, String> properties = new HashMap<>(config.getFileSystemProperties());
+    if (config.shouldCopyHeader()) {
+      properties.put(PathTrackingInputFormat.COPY_HEADER, "true");
     }
 
-    @Override
-    protected String getPath() {
-      return path;
+    // everything from this point on should be removed in a future release.
```

Comment:
Is there a jira for it? If so, it would be good to add it here.

Reply:
done